### ISSUE TITLE
Indicate default option with all steps

### ISSUE THREAD

**Issue by 89z** [rust-lang/rustup#2572]
Using this file:

https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-gnu/rustup-init.exe

I get this question, which presents with two choices, and marks the default
choice:

~~~
If you will be targeting the GNU ABI or otherwise know what you are
doing then it is fine to continue installation without the build
tools, but otherwise, install the C++ build tools before proceeding.

Continue? (Y/n)
~~~

Then I get this, which again marks the default:

~~~
   default host triple: x86_64-pc-windows-msvc
     default toolchain: stable (default)
               profile: default
  modify PATH variable: yes

1) Proceed with installation (default)
2) Customize installation
3) Cancel installation
~~~

However then I get this, which has a default but is not indicated:

~~~
I'm going to ask you the value of each of these installation options.
You may simply press the Enter key to leave unchanged.

Default host triple?
~~~

Then I get this, which has a default but is not indicated:

~~~
Default toolchain? (stable/beta/nightly/none)
~~~

Then I get this, which has a default but is not indicated:

~~~
Profile (which tools and data to install)? (minimal/default/complete)
~~~

Then I get this, which has a default but is not indicated:

~~~
Modify PATH variable? (y/n)
~~~

**Comment by kinnison** [comment]
Improving the prompting here ought to be fairly easy for someone looking to get into hacking on Rustup.

If you are interested in helping to fix this, please indicate so on the issue such that others don't duplicate your work.

If you require mentoring/assistance, ask on this issue, or want to discuss things before starting, you can find me or others on the `#wg-rustup` channel of the Rust discord server (https://discord.gg/rust-lang)

**Comment by a8nJ** [comment]
Currently I am downloading this:

https://static.rust-lang.org/dist/rust-1.48.0-x86_64-pc-windows-gnu.msi

and extracting with a program I wrote that extracts MSI files. Ideally what I
would like is just a Zip file of the above, but I dont think its available.

I think RustUp has the potential to be a good alternative, but the idea of
RustUp just throwing away 100 MB every time its run doesnt sit well with me.
So since that issue is a WONTFIX, I am unlikely to use RustUp going forward.

If you need to close the issue because I wont be trying to fix it myself, so be
it.

**Comment by kinnison** [comment]
@a8nJ I worry that you are doing something confusing with rustup if yo uthink it downloads 100MB every time it's run.  The rustup-init program will download a toolchain when you first install rustup, but after that rustup will use the installed toolchain unless you tell it to install another one.  Were you thinking `rustup-init.exe` was the interface to `rustup` always?

**Comment by a8nJ** [comment]
Sometimes a situation may arise where a users Rust installation becomes corrupted. Maybe a file is deleted, or renamed, or maybe something was added in error to the installation folder. Any number of reasons.

In those cases, the user is going to need to reinstall from the Tar archives. So it would make sense to use the ones that were already downloaded. But that cant happen, if RustUp just throws away the Tar files as soon as they are extracted.

**Comment by kinnison** [comment]
Thus far you are the only person who has reported such a likelihood.  I'm not saying it won't happen, but it seems unlikely to me.  It's far more likely in my opinion that every six weeks the user is going to need to update to the next stable which will necessitate downloading a new set of tarballs anyway.  Regardless, this particular issue was opened about prompts and indications of defaults, so perhaps we ought to stop this particular discussion for now.

**Comment by garethhubball** [comment]
Would anyone mind if I gave this a go please?

**Comment by kinnison** [comment]
Nominally this is fixed in #2605
